### PR TITLE
update iced

### PR DESCRIPTION
**Pull request by wash2**
seems to be working well , but I haven't been able to test with multiple monitors at my new apartment yet.

### CODE REVIEW

**Comment by ids1024** [comment]
Testing this (and not comparing to the previous version), when I run `cosmic-screenshot` then switch to window capture, it seems to show the background but not any window previews. When I re-run `cosmic-screenshot` it starts in that mode and does show windows.

**Comment by wash2** [comment]
> Testing this (and not comparing to the previous version), when I run cosmic-screenshot then switch to window capture, it seems to show the background but not any window previews. When I re-run cosmic-screenshot it starts in that mode and does show windows.

ahh ya, I think this is partially a rendering order issue.

File: src/screenshot.rs [hunk 1]
```diff
@@ -166,24 +159,46 @@ impl Screenshot {
...
+                            transform: Transform::_180,
+                            ..
+                        }) => imageops::rotate180(&img),
+                        Some(OutputInfo {
+                            transform: Transform::_270,
+                            ..
+                        }) => imageops::rotate270(&img),
+                        _ => img,
+                    };
+                    dbg!(img.width(), img.height());
```
Comment (ids1024) [inline src/screenshot.rs:191]: Presumably the `dbg!` call wasn't mean to be committed, and should be removed. (Or changed to something like a `log::debug!` with some more descriptive text).

File: src/screenshot.rs [hunk 2]
```diff
@@ -166,24 +159,46 @@ impl Screenshot {
...
+                    (img.width(), img.height(), img.into_vec().into())
+                })?,
+            );
         }
 
         Ok(map)
     }
 
     pub fn save_rgba(img: &RgbaImage, path: &PathBuf) -> anyhow::Result<()> {
+        dbg!(img.width(), img.height());
```
Comment (ids1024) [inline src/screenshot.rs:201]: Also this `dbg!` call.

File: src/screenshot.rs [hunk 3]
```diff
@@ -580,19 +594,30 @@ pub fn update_msg(portal: &mut CosmicPortal, msg: Msg) -> cosmic::Command<crate:
...
+                    if let Some((width, height, buf)) = images.remove(&name) {
                         if let Some(ref image_path) = image_path {
-                            if let Err(err) = Screenshot::save_rgba(&img, image_path) {
-                                log::error!("Failed to capture screenshot: {:?}", err);
-                            };
-                        } else {
-                            let mut buffer = Vec::new();
-                            if let Err(e) = Screenshot::save_rgba_to_buffer(&img, &mut buffer) {
-                                log::error!("Failed to save screenshot to buffer: {:?}", e);
+                            dbg!(width, height, buf.len());
```
Comment (ids1024) [inline src/screenshot.rs:599]: And this `dbg!` call.

**Comment by ids1024** [comment]
Screenshot and file chooser portals seem good.

When I try to screen cast (for instance with `obs`) I'm not seeing the dialog show up at all. While it seems to work on `master`.

**Review by ids1024 — APPROVED** [review]
Looks good and seems to work as expected now.

**Pull request merged**